src/Qt-RSA/qrsaencryption.cpp

```diff
@@ -387,17 +389,19 @@ QByteArray QRSAEncryption::decode(const QByteArray &rawData, const QByteArray &p
         default: return QByteArray();
     }
 }
-QByteArray QRSAEncryption::signMessage(QByteArray rawData, const QByteArray &privKey)
+QByteArray QRSAEncryption::signMessage(QByteArray rawData, const QByteArray &privKey,
+                                       QRSAEncryption::Rsa keyLength)
 {
     QByteArray hash = QCryptographicHash::hash(rawData, HashAlgorithm::Sha256);
 
-    QByteArray signature = QRSAEncryption(keyLength).encode(hash, privKey);
+    QByteArray signature = QRSAEncryption().encode(hash, privKey, keyLength);
```

Comment:
this method and not a function so it does not need extra objects
use     QByteArray signature = encode(hash, privKey, keyLength);